cvmfs/uid_map.h

```diff
@@ -110,7 +110,11 @@ class IntegerMap {
 
     return (HasDefault())
       ? default_value_
-      : T(0);
+      : k;
+  }
+
+  bool IsEmpty() const {
+    return map_.size() == 0 && !has_default_value_;
```

Comment:
Mhh, a map without rules but only a default value is still "empty", no? Since `::Map()` is now a NOOP for "empty" maps without default value, I would propose to add two methods instead:

* `::IsEmpty()` for any map with 0 rules
* `::HasEffect()` that returns `false` in the above case